### DISCUSSION TITLE
Issue While Trying to Use EJBCA Container Running on Docker

### DISCUSSION THREAD

**Discussion by Sarathkv99** [Keyfactor/ejbca-ce#289] [Q&A]
Hi, 
I'm facing an issue when i tried to use the EJBCA Container running on a Docker by following the videos provided in the Keyfactor YouTube channel. 

> ejbca-database  | 2023-05-25  5:48:02 533 [Warning] Access denied for user 'ejbca'@'172.19.0.3' (using password: YES)
> ejbca           | [ WARN] (main) Error: 1045-28000: Access denied for user 'ejbca'@'172.19.0.3' (using password: YES)
> ejbca           | 2023-05-25 05:48:02,631+0000 ERROR [org.ejbca.ui.cli.jdbc.JdbcTool] (main) (conn=533) Access denied for user 'ejbca'@'172.19.0.3' (using password: YES)
> ejbca           | 2023-05-25 05:48:02,632+0000 INFO  [org.ejbca.ui.cli.jdbc.JdbcTool] (main) Done.
> ejbca           | 2023-05-25 05:48:02,661+0000 INFO  [/opt/keyfactor/bin/start.sh] (process:1) Waiting for external database 'jdbc:mariadb://ejbca-database:3306/ejbca?characterEncoding=UTF-8' to become available.
> ejbca           | 2023-05-25 05:48:06,892+0000 INFO  [org.apache.commons.beanutils.FluentPropertyBeanIntrospector] (main) Error when creating PropertyDescriptor for public final void org.apache.commons.configuration2.AbstractConfiguration.setProperty(java.lang.String,java.lang.Object)! Ignoring this property.
> 
These were the logs i got and [[https://www.youtube.com/watch?v=oWC5vsGWXQ4&list=PLt17f5skfOPEcg-Hbn4d-YY22wdmnaEa9](https://www.youtube.com/watch?v=oWC5vsGWXQ4&list=PLt17f5skfOPEcg-Hbn4d-YY22wdmnaEa9)] this is the link of the youtube video that i followed.
Can anyone help me with this.
@netmackan 
@srasay2 
@hablutzel1 
@mlundblad 
@primetomas 
@svenska-primekey 
Thanks

**Comment by svenska-primekey** [comment]
I was looking in the issues for this from the email I got and now I finally found your question to respond. Can you share the docker-compose file you are using please. From reading what you shared the user account in MariaDB being used by EJBCA cannot authenticate to the DB.

**Reply by Sarathkv99** [reply]
Sure, here is the docker-compose file 

> version: '3'                                                                    
> networks:                                                                       
>   access-bridge:                                                                
>     driver: bridge                                                              
>   application-bridge:                                                           
>     driver: bridge                                                              
> services:                                                                       
>   ejbca-database:                                                               
>     container_name: ejbca-database                                              
>     image: "library/mariadb:latest"                                             
>     networks:                                                                   
>       - application-bridge                                                      
>     environment:                                                                
>       - MYSQL_ROOT_PASSWORD=ejbca                                             
>       - MYSQL_DATABASE=ejbca                                                    
>       - MYSQL_USER=ejbca                                                        
>       - MYSQL_PASSWORD=ejbca                                                     
>     volumes:                                                                    
>       - ./datadbdir:/var/lib/mysql:rw                                           
>   ejbca-node1:                                                                  
>     hostname: ejbca-node1                                                       
>     container_name: ejbca                                                       
>     image: keyfactor/ejbca-ce:latest                                            
>     depends_on:                                                                 
>       - ejbca-database                                                          
>     networks:                                                                   
>       - access-bridge                                                           
>       - application-bridge                                                      
>     environment:                                                                
>       - DATABASE_JDBC_URL=jdbc:mariadb://ejbca-database:3306/ejbca?characterEncoding=UTF-8
>       - LOG_LEVEL_APP=INFO                                                      
>       - LOG_LEVEL_SERVER=INFO                                                   
>       - TLS_SETUP_ENABLED=simple                                                
>     ports:                                                                      
>       - "80:8080"                                                               
>       - "443:8443"   

I did't get what you said last, can you help me more on it.

**Reply by svenska-primekey** [reply]
You need to add DATABASE_PASSWORD=(The MYSQL_PASSWORD VARIABLE) to the environment section of EJBCA if you are not using the default. The email notification I got you had a different password vvdn, and in the post here it now says ejbca... You can also exec to the database container and try a mysql -u ejbca -p ejbca to connect and make sure the password is correct for the account
